Marlin/src/inc/Conditionals_post.h

```diff
@@ -3306,10 +3306,11 @@
 
 // Number of VFAT entries used. Each entry has 13 UTF-16 characters
 #if ANY(SCROLL_LONG_FILENAMES, HAS_DWIN_E3V2, TFT_COLOR_UI)
-  #define MAX_VFAT_ENTRIES (5)
+  #define VFAT_ENTRIES_LIMIT 5
 #else
-  #define MAX_VFAT_ENTRIES (2)
+  #define VFAT_ENTRIES_LIMIT 2
 #endif
+#define MAX_VFAT_ENTRIES 20 // by VFAT specs to fit LFN of length 255
```

Comment:
Note that `VFAT_ENTRIES_LIMIT` is related to the number of characters that can be displayed on LCD and the methods that return the truncated filename to be displayed on LCD. So it should not necessarily be applied to the selection of files by long pathname with `M23` or to `M20` where we can use a smaller buffer to do the character comparisons. Once we have the selected file's ID and/or short pathname/filename, that is the only piece of data needed to refer to the selected file from that point onward, where we only need to store the complete long name of the selected file for reporting and display.

Meanwhile, the size of the buffer used by `SDCARD_SORT_ALPHA` is determined by `SDSORT_CACHE_VFATS`, where we might even store the sorted names in RAM. That might be optimized by just storing the sorted names in one contiguous buffer with a separate index into the buffer for each filename, rather than the current scheme of storing an index into an array of filenames (which mirrors the index into the directory structure used for the non-cached-filename sorting index).